src/preferences/node/json-preference-server.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import URI from '../../application/common/uri';
+import { FileSystem, FileChangesEvent, FileChangeType } from '../../filesystem/common/filesystem';
+import { FileSystemWatcher } from '../../filesystem/common/filesystem-watcher'
+import { IPreferenceClient } from '../common/preference-protocol'
+import { PreferenceChangedEvent } from '../common/preference-event'
+import { IPreferenceServer } from '../common/preference-protocol'
+import * as coreutils from "@phosphor/coreutils";
+
+export const PreferencePath = Symbol("PreferencePath")
+
+@injectable()
+export class JsonPreferenceServer implements IPreferenceServer {
+
+    protected prefs: { [key: string]: any } | undefined; // Preferences cache
+    protected client: IPreferenceClient | undefined;
+    // protected readonly resolveUri: Promise<URI>;
+
+    constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(FileSystemWatcher) protected readonly watcher: FileSystemWatcher,
+        @inject(PreferencePath) protected readonly preferencePath: URI) {
+
+        watcher.onFileChanges(event => {
+            if (this.arePreferencesAffected(event)) {
+                this.reconcilePreferences();
+            }
+        });
+
+        this.reconcilePreferences();
+    }
+
+    /**
+     * Checks to see if the preference file was modified
+     */
+    protected arePreferencesAffected(event: FileChangesEvent): boolean {
+        return event.changes.some(c => {
+            return (c.uri === this.preferencePath.toString() && c.type === FileChangeType.UPDATED);
```

Comment:
Yes you are right for the create event. I think I initially wrote that when testing the file events and the #192 bug was still there. I should change it. 

As for the delete event, I guess the prefs cache should be deleted as well. The default 
 pref server should back it up as the CompoundPrefServer will make sure that the defaults are called instead. WDYT?